File: 2024/AdventOfCode/Tests/Day09Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day09Test {
    private const string Example = "Inputs/09-Example.txt";
    private const string Input = "Inputs/09.txt";
    private static Day09 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day09(Example);
            const string expected = "1928";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day09(Input);
            const string expected = "6154342787400";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample() {
            _day = new Day09(Example);
            const string expected = "2858";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day09(Input);
            const string expected = "6183632723350";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}

File: 2025/test/advent_of_code/day_09_test.exs
defmodule AdventOfCode.Day09Test do
  use ExUnit.Case

  import AdventOfCode.Day09

  @day 9
  @year 2025

  @example String.trim("""
           7,1
           11,1
           11,7
           9,7
           9,5
           2,5
           2,3
           7,3
           """)

  test "part1 example" do
    result = part1(@example)

    assert result == 50
  end

  test "part1" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part1(input)

    assert result == 4_756_718_172
  end

  test "part2 example" do
    result = part2(@example)

    assert result == nil
  end

  test "part2" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part2(input)

    assert result == nil
  end
end
